src/wasi.rs

```diff
@@ -434,6 +434,175 @@ pub const NOEXPR: ::nl_item = 0x50001;
+
+        pub const PROT_NONE: ::c_int = 0;
+        pub const PROT_READ: ::c_int = 1;
+        pub const PROT_WRITE: ::c_int = 2;
+        pub const PROT_EXEC: ::c_int = 4;
+        pub const PROT_GROWSDOWN: ::c_int = 0x01000000;
+        pub const PROT_GROWSUP: ::c_int = 0x02000000;
+
+        pub const MS_ASYNC: ::c_int = 1;
+        pub const MS_INVALIDATE: ::c_int = 2;
+        pub const MS_SYNC: ::c_int = 4;
+
+        pub const MCL_CURRENT: ::c_int = 1;
+        pub const MCL_FUTURE: ::c_int = 2;
+        pub const MCL_ONFAULT: ::c_int = 4;
+
+        pub const POSIX_MADV_NORMAL: ::c_int = 0;
+        pub const POSIX_MADV_RANDOM: ::c_int = 1;
+        pub const POSIX_MADV_SEQUENTIAL: ::c_int = 2;
+        pub const POSIX_MADV_WILLNEED: ::c_int = 3;
+        pub const POSIX_MADV_DONTNEED: ::c_int = 4;
+
+        // #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
+        pub const MADV_NORMAL: ::c_int = 0;
+        pub const MADV_RANDOM: ::c_int = 1;
+        pub const MADV_SEQUENTIAL: ::c_int = 2;
+        pub const MADV_WILLNEED: ::c_int = 3;
+        pub const MADV_DONTNEED: ::c_int = 4;
+        pub const MADV_FREE: ::c_int = 8;
+        pub const MADV_REMOVE: ::c_int = 9;
+        pub const MADV_DONTFORK: ::c_int = 10;
+        pub const MADV_DOFORK: ::c_int = 11;
+        pub const MADV_MERGEABLE: ::c_int = 12;
+        pub const MADV_UNMERGEABLE: ::c_int = 13;
+        pub const MADV_HUGEPAGE: ::c_int = 14;
+        pub const MADV_NOHUGEPAGE: ::c_int = 15;
+        pub const MADV_DONTDUMP: ::c_int = 16;
+        pub const MADV_DODUMP: ::c_int = 17;
+        pub const MADV_WIPEONFORK: ::c_int = 18;
+        pub const MADV_KEEPONFORK: ::c_int = 19;
+        pub const MADV_COLD: ::c_int = 20;
+        pub const MADV_PAGEOUT: ::c_int = 21;
+        pub const MADV_HWPOISON: ::c_int = 100;
+        pub const MADV_SOFT_OFFLINE: ::c_int = 101;
+        // #endif
+
+        // #ifdef _GNU_SOURCE
+        pub const MREMAP_MAYMOVE: ::c_int = 1;
+        pub const MREMAP_FIXED: ::c_int = 2;
+        pub const MREMAP_DONTUNMAP: ::c_int = 4;
+
+        pub const MLOCK_ONFAULT: ::c_uint = 0x01;
+
+        pub const MFD_CLOEXEC: ::c_uint = 0x0001;
+        pub const MFD_ALLOW_SEALING: ::c_uint = 0x0002;
+        pub const MFD_HUGETLB: ::c_uint = 0x0004;
+        // #endif
+
+        // Included for future reference (for mmap64):
+        // #if defined(_LARGEFILE64_SOURCE) || defined(_GNU_SOURCE)
+        // #define off64_t off_t
+        // #define mmap64 mmap
+        // #endif
+
+        #[cfg_attr(
+            feature = "rustc-dep-of-std",
+            link(
+                name = "c",
+                kind = "static",
+                modifiers = "-bundle",
+                cfg(target_feature = "crt-static")
+            )
+        )]
+        #[cfg_attr(
+            feature = "rustc-dep-of-std",
+            link(name = "c", cfg(not(target_feature = "crt-static")))
+        )]
```

Comment:
I grabbed this from the `extern "C"` block later in the file. As far as I can tell, this just specifies how to handle static linking (adding the "bundle" modifier) when compiling to `crt-static` when building as a dep of std, so I copied it to this new extern block.
If this is wrong, please let me know!